Source/common/santa.proto

```diff
@@ -538,6 +538,229 @@ message FileAccess {
   optional PolicyDecision policy_decision = 6;
 }
 
+////////////////////
+//
+// New Shared Types
+//
+////////////////////
+
+message GraphicalSession {
+  optional uint32 id = 1;
+}
+
+message Address {
+  optional bytes address = 1;
+
+  enum AddressType {
+    ADDRESS_TYPE_UNKNOWN = 0;
+    ADDRESS_TYPE_IPV4 = 1;
+    ADDRESS_TYPE_IPV6 = 2;
+    ADDRESS_TYPE_NAMED_SOCKET = 3;
+  }
+
+  AddressType type = 2;
+}
+
+////////////////////
+//
+// START OPTION 1
+//
+////////////////////
+
+message LoginWindowSessionLogin {
+  optional UserInfo user = 1;
+  optional GraphicalSession graphical_session = 2;
+}
+
+message LoginWindowSessionLogout {
+  optional UserInfo user = 1;
+  optional GraphicalSession graphical_session = 2;
+}
+
+message LoginWindowSessionLock {
+  optional UserInfo user = 1;
+  optional GraphicalSession graphical_session = 2;
+}
+
+message LoginWindowSessionUnlock {
+  optional UserInfo user = 1;
+  optional GraphicalSession graphical_session = 2;
+}
+
+message LoginWindowSession {
+  oneof event {
+    LoginWindowSessionLogin login = 1;
+    LoginWindowSessionLogout logout = 2;
+    LoginWindowSessionLock lock = 3;
+    LoginWindowSessionUnlock unlock = 4;
+  }
+}
+
+message LoginLogoutLogin {
+  optional bool success = 1;
+  optional bytes failure_message = 2;
+  optional UserInfo user = 3;
+}
+
+message LoginLogoutLogout {
+  optional UserInfo user = 1;
+}
+
+message LoginLogout {
+  oneof event {
+    LoginLogoutLogin login = 1;
+    LoginLogoutLogout logout = 2;
+  }
+}
+
+message ScreenSharingAttach {
+  optional Address source = 1;
+  optional bytes viewer = 2;
+  optional bytes authentication_type = 3;
+  optional UserInfo authentication_user = 4;
+  optional UserInfo session_user = 5;
+  optional bool existing_session = 6;
+  optional GraphicalSession graphical_session = 7;
+}
+
+message ScreenSharingDetach {
+  optional Address source = 1;
+  optional bytes viewer = 2;
+  optional GraphicalSession graphical_session = 3;
+}
+
+message ScreenSharing {
+  oneof event {
+    ScreenSharingAttach attach = 1;
+    ScreenSharingDetach detach = 2;
+  }
+}
+
+message OpenSSHLogin {
```

Comment:
I think SSHLogin, SSHLogout and SSH (or SSHEvent) would be good here, the "Open" feels needless

Reply:
Mentioning for posterity - this is specific to OpenSSH that ships on the system, not SSH generically. I don't have a strong preference here and think its fair to drop the prefix, but there is a chance for confusion. This is maybe more just a note to make sure we're clear in documentation about the limitation of this event.

Reply:
Sticking to OpenSSH is fine, I retract my comment